packages/page-staking/src/Payouts/index.tsx

```diff
@@ -150,11 +150,19 @@ function getOptions (api: ApiPromise, eraLength: BN | undefined, historyDepth: B
   return [{ text: '', value: 0 }];
 }
 
+function getMyStashesIndex (api: ApiPromise, hasOwnValidators: boolean) {
+  return (isFunction(api.tx.staking.payoutStakers) && hasOwnValidators) ? 0 : 1;
+}
+
 function Payouts ({ className = '', isInElection, ownValidators }: Props): React.ReactElement<Props> {
   const { t } = useTranslation();
   const { api } = useApi();
-  const [hasOwnValidators] = useState(() => ownValidators.length !== 0);
-  const [myStashesIndex, setMyStashesIndex] = useState(() => (isFunction(api.tx.staking.payoutStakers) && hasOwnValidators) ? 0 : 1);
+  const hasOwnValidators = useMemo(() => ownValidators.length !== 0, [ownValidators]);
+  const [myStashesIndex, setMyStashesIndex] = useState(() => getMyStashesIndex(api, hasOwnValidators));
+
+  useEffect(() => {
+    setMyStashesIndex(getMyStashesIndex(api, hasOwnValidators));
+  }, [api, hasOwnValidators]);
```

Comment:
```suggestion
  }, [api, hasOwnValidators]);

```

Weirdly the linter didn't complain here...

Reply:
sure